---
title: "Supplementary Figures and Tables"
author: 
- Selcen Ari
- Alper Yilmaz 
date: "`r Sys.Date()`"
output:
  pdf_document: default
  html_document: default
  word_document: default
bibliography: NAR-sample.bib
urlcolor: blue
header-includes: 
  \usepackage{float} \floatplacement{figure}{H} 
  \newcommand{\beginsupplement}{\setcounter{table}{0}  \renewcommand{\thetable}{S\arabic{table}} \setcounter{figure}{0}     \renewcommand{\thefigure}{S\arabic{figure}}}
  \usepackage{subfig}
  \usepackage{tabularx}
  \usepackage{caption}
---
\tableofcontents

\beginsupplement


# 2. Supplementary Figures 

![Workflow for simulation of competing endogenous RNA regulations. Graph object in steps 2-6 is saved and updated continuously.](workflow1.png){#figs1}

```{r, include=FALSE, message=FALSE, warning=FALSE}

library(tidygraph)
library(dplyr)
library(tidyr)
library(stringr)
library(ggraph)
library(ggplot2)
knitr::opts_chunk$set(tidy.opts=list(width.cutoff=50),tidy=TRUE)
#knitr::opts_chunk$set(fig.pos = 'H')
knitr::opts_chunk$set(cache=FALSE)

```

```{r, message=FALSE, warning=FALSE, echo=FALSE}
library(ceRNAnetsim)
```

## 2.1 Sample dataset analysis in absence of interaction factors.

```{r, echo=FALSE}
# Sample dataset refers to minsamp in ceRNAnetsim package.
data("minsamp")
```


```{r, message=FALSE, warning=FALSE,echo=FALSE, fig.cap= "Sample Dataset in Steady-state. InÄ±tial expression levels in minsamp network (*Sample* network in manuscript). The network contains two miRNAs and 6 genes with arbitrarily chosen expression values. Refer to Table S1 for exact expression values", fig.width=5, fig.height=4}
priming_graph(minsamp, competing_count = Competing_expression, miRNA_count = miRNA_expression)%>%
  vis_graph(Competing_color = "navajowhite3", mirna_color = "ivory4",
            title = " ")+
  geom_node_text(aes(label = name), repel = TRUE, size =3)
```


```{r, message=FALSE, warning=FALSE,echo=FALSE,  fig.keep='all', fig.cap= "Gene2 Upregulation on Sample Dataset. Expression level of Gene2 is increased from 10,000 to 20,000 in order to demonstrate the calculation steps.", fig.width=5, fig.height=4}
priming_graph(minsamp, competing_count = Competing_expression, miRNA_count = miRNA_expression)%>%
  update_how("Gene2", 2)%>%
  vis_graph(Competing_color = "navajowhite3", mirna_color = "ivory4", Upregulation = "red", 
            title = " ")+
  geom_node_text(aes(label = name), repel = TRUE, size =3)
```

```{r, message=FALSE, warning=FALSE, echo=FALSE, out.width='50%', fig.asp=1, fig.ncol = 2,  fig.show='hold', fig.cap = "Sequential iteration of Sample data after up-regulation of Gene2. A) First response of system to Gene2 upregulation (2nd iteration). B) Spreading of perturbation on system (3th iteration)", fig.subcap= NULL}

priming_graph(minsamp, 
              competing_count = Competing_expression, 
              miRNA_count = miRNA_expression) %>%
  update_how("Gene2", 2) %>%
  simulate(cycle=1) %>%
  vis_graph(Competing_color = "navajowhite3", 
            mirna_color = "ivory4", 
            Upregulation = "red", 
            Downregulation = "blue", 
            title = "A")+
  geom_node_text(aes(label = name), repel = TRUE, size =3)

priming_graph(minsamp, 
              competing_count = Competing_expression, 
              miRNA_count = miRNA_expression)%>%
  update_how("Gene2", 2)%>%
  simulate(2)%>%
  vis_graph(Competing_color = "navajowhite3", 
            mirna_color = "ivory4", 
            Upregulation = "red", 
            Downregulation = "blue", 
            title = "B")+
  geom_node_text(aes(label = name), repel = TRUE, size =3)

```

## 2.2 Calculations with interaction factors

```{r, echo=FALSE, fig.cap="Calculation of initial miRNA repression activity using interaction parameters in Sample+ network. A) Interaction paramaters between genes and miRNAs in Sample+ are shown on network while expression levels can be found in Table S1. B) Interaction paramaters were updated after normalisation C) Amount of miRNA distributed to each mRNA according to mRNA levels and affinity parameters are shown on edges. D) Values on edges indicate repression activity. Red values indicate repression activity affected by region effect (RE) parameter. E) Total repression on G4 from two miRNAs is calculated by adding repression values originating from both miRNAs.", fig.show='hold', out.height= "70%", out.width="70%", fig.align = "center"}
knitr::include_graphics("fig_s5.png")

```


$G$, Gene; $M$, miRNA; $STE$, seed type effect; $RE$, Region Effect; $E$, Energy; $STE^\prime$, normalized values of seed type efficiency coefficient; $RE^\prime$, normalized values of region efficiency coefficient; $E^\prime$, normalized values of energy coefficient. Numbering on edges match the pair order in Table S1.

## 2.3 Sample+ dataset analysis with interaction factors.

```{r, message=FALSE, warning=FALSE, echo=FALSE, fig.cap= "Sample+ in Steady-state. Interaction factors of Sample+ network are available in Table S1.", fig.width=5, fig.height=4}
priming_graph(minsamp, competing_count = Competing_expression, miRNA_count = miRNA_expression, aff_factor = c(energy, seed_type), deg_factor = region)%>%
  vis_graph(Competing_color = "navajowhite3", mirna_color = "ivory4", 
            title = " ")+
  geom_node_text(aes(label = name), repel = TRUE, size =3)

```

```{r, message=FALSE, warning=FALSE, echo=FALSE,  fig.cap= "Perturbation in Sample+ network by two-fold increase in Gene2 expression level.", fig.width=5, fig.height=4}
priming_graph(minsamp, competing_count = Competing_expression, miRNA_count = miRNA_expression, aff_factor = c(energy, seed_type), deg_factor = region)%>%
  update_how("Gene2", 2)%>%
  vis_graph(Competing_color = "navajowhite3", mirna_color = "ivory4", Upregulation = "red", 
            title = " ")+
  geom_node_text(aes(label = name), repel = TRUE, size =3)
```

```{r, message=FALSE, warning=FALSE,echo=FALSE, out.width='50%', fig.asp=1, fig.ncol = 2, fig.show='hold', fig.cap = "Sequential iteration of Sample+ A) First response of system to Gene2 upregulation (2nd iteration). B) Spreading of perturbation on system (3rd iteration). Although visualisation looks similar to Figure S4B, current counts of genes are drastically different."}
priming_graph(minsamp, competing_count = Competing_expression, miRNA_count = miRNA_expression, aff_factor = c(energy, seed_type), deg_factor = region)%>%
  update_how("Gene2", 2)%>%
  simulate(cycle = 1)%>%
  vis_graph(Competing_color = "navajowhite3", mirna_color = "ivory4", Upregulation = "red", Downregulation = "blue", title = "A")+
  geom_node_text(aes(label = name), repel = TRUE, size =3)

priming_graph(minsamp, competing_count = Competing_expression, miRNA_count = miRNA_expression, aff_factor = c(energy, seed_type), deg_factor = region)%>%
  update_how("Gene2", 2)%>%
  simulate(cycle = 2)%>%
  vis_graph(Competing_color = "navajowhite3", mirna_color = "ivory4", Upregulation = "red", Downregulation = "blue", title = "B")+
  geom_node_text(aes(label = name), repel = TRUE, size =3)
  
  
```

## 2.4 Common target perturbation in Sample+ dataset.

```{r, message=FALSE, warning=FALSE, echo=FALSE, out.width='50%', fig.asp=1, fig.ncol = 2, fig.show='hold', fig.cap = "Perturbation of Gene4 and its effects on Sample+. A) Network at steady-state. B) Upregulation of Gene4. C) Primary response of network to upregulation of Gene4. D) Re-regulation of whole nodes on system (3th iteration)"}

priming_graph(minsamp, competing_count = Competing_expression, miRNA_count = miRNA_expression, aff_factor = c(energy, seed_type), deg_factor = region)%>%
  vis_graph(Competing_color = "navajowhite3", mirna_color = "ivory4", Upregulation = "red", title = "A")+
  geom_node_text(aes(label = name), repel = TRUE, size =3)

priming_graph(minsamp, competing_count = Competing_expression, miRNA_count = miRNA_expression, aff_factor = c(energy, seed_type), deg_factor = region)%>%
  update_how("Gene4", 2)%>%
  vis_graph(Competing_color = "navajowhite3", mirna_color = "ivory4", Upregulation = "red", title = "B")+
  geom_node_text(aes(label = name), repel = TRUE, size =3)

priming_graph(minsamp, competing_count = Competing_expression, miRNA_count = miRNA_expression, aff_factor = c(energy, seed_type), deg_factor = region)%>%
  update_how("Gene4", 2)%>%
  simulate(cycle=1)%>%
  vis_graph(Competing_color = "navajowhite3", mirna_color = "ivory4", Upregulation = "red", title = "C")+
  geom_node_text(aes(label = name), repel = TRUE, size =3)

priming_graph(minsamp, competing_count = Competing_expression, miRNA_count = miRNA_expression, aff_factor = c(energy, seed_type), deg_factor = region)%>%
  update_how("Gene4", 2)%>%
  simulate(cycle=2)%>%
  vis_graph(Competing_color = "navajowhite3", mirna_color = "ivory4", Upregulation = "red", title = "D")+
  geom_node_text(aes(label = name), repel = TRUE, size =3)
```



![Sankey diagram represents top five KEGG and GO (molecular function and biological process) terms and genes enriched on these terms. Genes with single edge are not shown.](sankey_v2.png){#figs9}


```{r, message=FALSE, warning=FALSE, echo=FALSE, fig.width=12, fig.height=5,fig.show='hold', fig.cap = "Log2 transformed expression levels of tumor-specific perturbing nodes in tumor and normal tissue samples of 87 patients."}

matched_patient_exp_dataset <- readRDS("figures_manuscript/matched_patient_exp_dataset.RDS")
mirtarbase_gene_res <- readRDS("figures_manuscript/mirtarbase_gene_res.RDS")

matched_patient_exp_dataset%>%
  filter(Ensembl_Gene_Id %in% pull(filter(mirtarbase_gene_res, situ == "tumor_specific"), Ensembl_Gene_Id))%>%
  pivot_longer(cols = contains("TCGA"), names_to="patient_id", values_to = "expressions")%>%
  mutate(patient = ifelse(endsWith(patient_id, "01A"), "tumor", "normal"))%>%
  left_join(mirtarbase_gene_res, by ="Ensembl_Gene_Id")%>%
  group_by(patient, Ensembl_Gene_Id)%>%
  ggplot(aes(x= hgnc_symbol, y=log2(expressions), fill = patient))+
  #scale_y_log10()+
  geom_boxplot(position = "dodge")+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 90), panel.background = element_rect())+
  xlab("")
```


# 3 Supplementary Tables

```{r, eval=FALSE, include=FALSE}
library(xtable)
knitr::opts_chunk$set(tidy.opts=list(width.cutoff=50),tidy=TRUE)
library(gt)
knitr::opts_chunk$set(cache=FALSE)

```

## 3.1 *Sample+* dataset


|Competing| miRNA | Competing Expression | miRNA Expression | Seed Type Effect | Region Effect | Energy
|-----------|------------|----------|----------------|---------|-----------|----------|
| `Gene1`   | `Mir1` | 10000 | 1000 | 0.43 | 0.30 | -20 |
| `Gene2`   | `Mir1` | 10000 | 1000 | 0.43 | 0.01 | -15 |
| `Gene3`   | `Mir1` | 5000  | 1000 | 0.32 | 0.40 | -14 |
| `Gene4`   | `Mir1` | 10000 | 1000 | 0.23 | 0.50 | -10 |
| `Gene4`   | `Mir2` | 10000 | 2000 | 0.35 | 0.90 | -12 |
| `Gene5`   | `Mir2` | 5000  | 2000 | 0.05 | 0.40 | -11 |
| `Gene6`   | `Mir2` | 10000 | 2000 | 0.01 | 0.80 | -25 |

Table: The parameters which affect miRNA:target interactions (i.e. seed type, region, energy) are provided in Sample+ dataset, while these factors are not utilized in simulation of Sample dataset.

## 3.2 Significant factors in miRNA:target interactions

Some of information about miRNA:target interactions were exhibited directly by high-throughput studies. On the other hand, we were examined other interacion factors based on different studies.

+ [@helwak_mapping_2013; @moore_mirnatarget_2015] reported the energy values in miRNA:target interactions.
+ Comparisons of canonical seed types were evaluated by study of [@grimson_microrna_2007], while functional and non-functional seed interactions were studied by [@bartel_micrornas:_2009] and [@betel2010comprehensive]. 
+ Numeric definition of target region location effect was performed based on studies of [@hausser_analysis_2013] and [@helwak_mapping_2013]



|seed type | seed type effect 
|----------|---------------|
| 6-mer_noncanonical | 0.05 | 
| 9-mer | 0.43 | 
| 6-mer | 0.07 | 
| 8-mer | 0.43 | 
| 7-mer | 0.23 | 
| none | 0.01 | 
| 5-mer_noncanonical | 0.04 | 
| 5-mer | 0.05 | 
| 6-merA1_noncanonical | 0.05 | 
| 7-mer-8m_noncanonical | 0.21 | 
| 7-mer-8m | 0.25 | 
| 8-mer_noncanonical | 0.35 | 
| 7-merA1_noncanonical | 0.16 | 
| 7-merA1 | 0.19 | 
| 6-merA1 | 0.07 | 

Table: Efficiency factors for seed types.


| region | region effect
|---------|-------------|
| 3UTR | 0.84 |
| CDS | 0.42 | 
| 3UTRCDS | 0.93 | 
| 5UTR | 0.01 |
| 5UTRCDS | 0.42 | 
| none  | 0.01 | 
| intron | 0.01 | 
| CDS3UTR | 0.93 | 
| CDS5UTR | 0.42 | 
| exon_unclassified | 0.20 | 
| CDS3UTRintron | 0.93 | 
| 3UTRintron | 0.84 | 
| CDSintron | 0.42 | 
| 5UTRintron | 0.01 | 
| 5UTR3UTR | 0.93 | 
| CDS5UTR3UTR | 0.93 |

Table: Efficiency factors for binding regions on targets


## 3.3 Content of High-throughput experimental studies 

|  Variable | Definition 
|-----------|-------|
|  cluster    | Barcode from experimental method |
|  chromosome    | Chromosome of Target gene from raw data |
|   start_position   | Gene start position from raw data |
|   end_position    | Gene end position from raw data |
|   strand    | Gene strand |
|   hgnc_symbol     | Gene name (Symbol) |
|   Ensembl_Gene_Id     | Ensembl Gene Id of gene |
|   Ensembl_Transcript_Id   | Ensembl transcript id of mRNA of Target gene |
|   target_seq   | mRNA sequences targeted by miRNA |
|   miRNA   | miRNA id (from miRBase version 21 ) |
|   miR_seq   | miRNA sequence |
|   seed_type  | seed type of miRNA:target interaction |
|   Energy   | Energy of miRNA:target binding |
|   HG38build_loc    | Recent chromosomal location of Gene |
|   Genome_build   | Genome build of given chromosome, start and end positions |
|   region   | interaction location on target | 
|   region_effect   | Coefficient of location on target | 
|   seed_type_effect | Coefficient for seed sequence of miRNA:target interaction |

Table: Context of miRNA:target pairs supported by High-throughput Experiments. CLEAR-CLiP and CLASH datasets were integrated as described in Section 2 of Supplementary Material and Method.


## 3.4  Variables of network object during simulation

As a result of simulation a dataset, a graph object is obtained that includes various variables in edge and node data. A graph object includes variables at Table S5.

 \begin{table}[ht]
\centering
\caption{ The context graph object during the process.}
\begin{tabularx}{\textwidth}{lX}
  \hline
 Variables & Description \\
  \hline
   \emph{Node Variables} &  \\
   name    &  node name \\
   type    & Competing or miRNA \\
   node\_id    &  in on graph object \\
   initial\_count     & Initial Expression value of node \\
   count\_pre     &  Expression value of node at previous regulation \\
   count\_current     &  Existing expression value of node \\
   changes\_variable     &  Regulation of node (Up, down or steady) \\
   \hline
   \emph{Edge Variables} & \\
   Competing name   &  name of genes\\
   miRNA name   &  name of miRNAs \\
   Competing expression   &  Expression values of competing elements at steady-state \\
   miRNA expression   &  Expression values of miRNA elements at steady-state \\
   energy   & coefficient of miRNA:target interactions (binding affinity) \\
   seed type   &  coefficient of miRNA:target interactions (binding affinity) \\
   region   &  coefficient of miRNA:target interactions (degradation efficiency) \\
   afff factor   &  coefficient scaled and combined affinity factor \\
   degg factor   & coefficient scaled and combined degradation factor \\ 
   comp\_count\_list   & list of competing expression for each iteration \\ 
   comp\_count   & pre: competing expression at previous iteration; current: competing expression at present iteration \\
   mirna\_count\_list   & list of miRNA expression for each iteration \\
   mirna\_count   &  pre: miRNA expression at previous iteration; current: miRNA expression at present iteration \\
   effect   & pre: total miRNA reppressive effect on individual target at previous iteration ; current: miRNA reppressive effect on individual   target at present iteration \\
   effect\_list   & list of miRNA reppressive effect on individual target for each iteration \\
  \hline
\end{tabularx}
\end{table}


# 4. Access to code

All codes are available at [github repo](https://github.com/selcenari/ceRNA_multifactorial_model).

\newpage

# REFERENCES

