---
title: "Supplementary Tables"
author: 
- Selcen Ari
- Alper Yilmaz 
date: "16 09 2019"
output:
  pdf_document: default
  html_document: default
  word_document: default
bibliography: ref.bib
header-includes: 
  \usepackage{tabularx}
  \usepackage{float}
  \usepackage{caption}
  \newcommand{\beginsupplement}{\setcounter{table}{0}  \renewcommand{\thetable}{S\arabic{table}} \setcounter{figure}{0} \renewcommand{\thefigure}{S\arabic{figure}}}
---


\beginsupplement


```{r, eval=FALSE, include=FALSE}
library(xtable)
knitr::opts_chunk$set(tidy.opts=list(width.cutoff=50),tidy=TRUE)
library(gt)

```


|competing| miRNA | Competing_expression | miRNA_expression | seed_type | region | energy
|-----------|------------|----------|----------------|---------|-----------|----------|
| `Gene1`   | `Mir1` | 10000 | 1000 | 0.43 | 0.30 | -20 |
| `Gene2`   | `Mir1` | 10000 | 1000 | 0.43 | 0.01 | -15 |
| `Gene3`   | `Mir1` | 5000  | 1000 | 0.32 | 0.40 | -14 |
| `Gene4`   | `Mir1` | 10000 | 1000 | 0.23 | 0.50 | -10 |
| `Gene4`   | `Mir2` | 10000 | 2000 | 0.35 | 0.90 | -12 |
| `Gene5`   | `Mir2` | 5000  | 2000 | 0.05 | 0.40 | -11 |
| `Gene6`   | `Mir2` | 10000 | 2000 | 0.01 | 0.80 | -25 |

Table: *minsamp* sample dataset that includes interaction factors. The parameters which affect miRNA:target interactions (i.e. seed type, region, energy) are provided in sample dataset.


Note: Energy values in miRNA:target pairs are represented by high-throughput studies [@helwak_mapping_2013; @moore_mirnatarget_2015] which are utilized in this study. On the other hand, we have specified the other interaction factors, seed type and location of binding region on the target, as numeric values based on the previous studies.[@grimson_microrna_2007] have compared the seed types' effect on target repression with few miRNA had canonical seed pairing in their study. Additionally, [@bartel_micrornas:_2009] and [@betel2010comprehensive] have studied on functional and non-functional seed interactions. Based on results of these studies we have arranged seed types of miRNA:target interactions as numeric values. We also have redefined location of binding region on the target as numeric values, based on studies of [@hausser_analysis_2013] and [@helwak_mapping_2013]. With this process, we have handled this integrated dataset in context of competitor behaviors and functionality of interactions. 


\begin{table}[ht]
\centering
\caption{Efficiency factors for seed types.}
\begin{tabular}{rlr}
  \hline
 & seed\_type & seed\_type\_effect \\ 
  \hline
1 & 6-mer\_noncanonical & 0.05 \\ 
  2 & 9-mer & 0.43 \\ 
  3 & 6-mer & 0.07 \\ 
  4 & 8-mer & 0.43 \\ 
  5 & 7-mer & 0.23 \\ 
  6 & none & 0.01 \\ 
  7 & 5-mer\_noncanonical & 0.04 \\ 
  8 & 5-mer & 0.05 \\ 
  9 & 6-merA1\_noncanonical & 0.05 \\ 
  10 & 7-mer-8m\_noncanonical & 0.21 \\ 
  11 & 7-mer-8m & 0.25 \\ 
  12 & 8-mer\_noncanonical & 0.35 \\ 
  13 & 7-merA1\_noncanonical & 0.16 \\ 
  14 & 7-merA1 & 0.19 \\ 
  15 & 6-merA1 & 0.07 \\ 
   \hline
\end{tabular}
\end{table}


\begin{table}[ht]
\centering
\caption{ Efficiency factors for binding regions on targets}
\begin{tabular}{rlr}
  \hline
 & region & region\_effect \\ 
  \hline
1 & 3UTR & 0.84 \\ 
  2 & CDS & 0.42 \\ 
  3 & 3UTRCDS & 0.93 \\ 
  4 & 5UTR & 0.01 \\ 
  5 & 5UTRCDS & 0.42 \\ 
  6 &  & 0.01 \\ 
  7 & intron & 0.01 \\ 
  8 & CDS3UTR & 0.93 \\ 
  9 & CDS5UTR & 0.42 \\ 
  10 & exon\_unclassified & 0.20 \\ 
  11 & CDS3UTRintron & 0.93 \\ 
  12 & 3UTRintron & 0.84 \\ 
  13 & CDSintron & 0.42 \\ 
  14 & 5UTRintron & 0.01 \\ 
  15 & 5UTR3UTR & 0.93 \\ 
  16 & CDS5UTR3UTR & 0.93 \\ 
   \hline
\end{tabular}
\end{table}


|Hugo_Symbol| miRNA_name | mirna_RPM | GE_normal | energy | seed_type_effect | region_effect |
|-----------|------------|----------|----------------|---------|-----------|----------|
| ENAH | hsa-let-7a-5p | 111204.15 | 7540 | -22.70 | 0.01 | 0.00 |
| GALNT2 | hsa-let-7a-5p | 111204.15 | 2824 | -20.50 | 0.14 | 0.00 | 
| RLF | hsa-let-7a-5p | 111204.15 | 1144 | -18.00 | 0.03 | 0.20 |
| MAST2 | hsa-let-7a-5p | 111204.15 | 2640 | -22.60 | 0.24 | 0.00 | 
| DOCK1 | hsa-let-7a-5p | 111204.15 | 4826 | -20.50 | 0.01 | 0.00 | 
| ZBTB16 | hsa-let-7a-5p | 111204.15 | 315 | -24.30 | 0.14 | 0.00 |

Table: Example of E9GE_mirnagenenormal dataset.


 \begin{table}[ht]
\centering
\caption{ miRNA:target pairs supported by High-throughput Experiments}
\begin{tabular}{lll}
  \hline
 Variable & Structure & Means \\
  \hline
   cluster    & character & Barcode from experimentally method \\
   chromosome    & character & Chromosome of Target gene from raw data \\
   start\_position    & numeric & Gene start position from raw data \\
   end\_position     & numeric & Gene end position from raw data \\
   strand     & character & Gene strand \\
   hgnc\_symbol     & character & Gene name (Symbol) \\
   Ensembl\_Gene\_Id     & character & Ensembl Gene Id of gene \\
   Ensembl\_Transcript\_Id   & character & Ensembl transcript id of mRNA of Target gene \\
   target\_seq   & character & mRNA sequences targeted by miRNA \\
   miRNA   & character & miRNA id (from miRBase version 21 ) \\
   miR\_seq   & character & miRNA sequence \\
   seed\_type   & character & seed type of miRNA:target interaction \\
   Energy   & numeric & Energy of miRNA:target binding \\
   HG38build\_loc   & character & Recent chromosomal location of Gene \\
   Genome\_build   & character& Genome build of given chromosome, start and end positions \\
   region   & character & interaction location on target \\ 
   region\_effect   & numeric & Coefficient of location on target \\ 
   seed\_type\_effect   & numeric & Coefficient for seed sequence of miRNA:target interaction \\
   \hline
\end{tabular}
\end{table}


As a result of simulation a dataset, a graph object is obtained that includes various variables in edge and node data. A graph object includes variables at following.

 \begin{table}[ht]
\centering
\caption{ The context graph object during the process.}
\begin{tabularx}{\textwidth}{llX}
  \hline
 Variables & Structure & Means \\
  \hline
   \emph{Node Variables} & & \\
   name    & character & node name \\
   type    & character &Competing or miRNA \\
   node\_id    & numeric & in on graph object \\
   initial\_count     & numeric & Initial Expression value of node \\
   count\_pre     & numeric & Expression value of node at previous regulation \\
   count\_current     & numeric & Existing expression value of node \\
   changes\_variable     & character & Regulation of node (Up, down or steady) \\
   \hline
   \emph{Edge Variables} & & \\
   Competing name   & character & name of genes\\
   miRNA name   & character & name of miRNAs \\
   Competing expression   & numeric & Expression values of competing elements at steady-state \\
   miRNA expression   & numeric & Expression values of miRNA elements at steady-state \\
   energy   & numeric & coefficient of miRNA:target interactions (binding affinity) \\
   seed type   & numeric & coefficient of miRNA:target interactions (binding affinity) \\
   region   & numeric & coefficient of miRNA:target interactions (degradation efficiency) \\
   afff factor   & numeric& coefficient scaled and combined affinity factor \\
   degg factor   & numeric &coefficient scaled and combined degradation factor \\ 
   comp\_count\_list   & list & list of competing expression for each iteration \\ 
   comp\_count   & numeric & pre: competing expression at previous iteration; current: competing expression at present iteration \\
   mirna\_count\_list   & list & list of miRNA expression for each iteration \\
   mirna\_count   & numeric & pre: miRNA expression at previous iteration; current: miRNA expression at present iteration \\
   effect   & numeric & pre: total miRNA reppressive effect on individual target at previous iteration ; current: miRNA reppressive effect on individual   target at present iteration \\
   effect\_list   & list & list of miRNA reppressive effect on individual target for each iteration \\
  \hline
\end{tabularx}
\end{table}

\newpage


## REFERENCES
